src/main/java/io/cdap/plugin/batch/source/FTPBatchSource.java

```diff
@@ -317,5 +321,79 @@ private void validateAuthenticationInPath(FailureCollector collector) {
         collector.getOrThrowException();
       }
     }
+
+    /**
+     * This method checks if user enters a valid ftp path by first checking the server
+     * and port number, then the username and password. Will timeout after 10 seconds
+     * if connection is not established.
+     *
+     * @param collector
+     */
+    void validateConnection(FailureCollector collector) {
+      Path urlInfo;
+      String extractedPassword = extractPasswordFromUrl();
+      String encodedPassword = URLEncoder.encode(extractedPassword);
+      String validatePath = path.replace(extractedPassword, encodedPassword);
+      try {
+        urlInfo = new Path(validatePath);
+      } catch (Exception e) {
+        throw new IllegalArgumentException(String.format("Unable to parse url: %s %s", e.getMessage(), e));
+      }
+      // server is the substring between @ and : in the path
+      String server = urlInfo.toUri().getAuthority().substring(urlInfo.toUri().getAuthority().lastIndexOf("@") + 1,
+              urlInfo.toUri().getAuthority().lastIndexOf(":"));
+      String user = urlInfo.toUri().getAuthority().split(":")[0];
+      String protocol = urlInfo.toUri().getScheme();
+      int port = urlInfo.toUri().getPort();
+      if (port == -1 && protocol.equals(FTP_PROTOCOL)) {
+        port = DEFAULT_FTP_PORT;
+      }
+      if (port == -1 && protocol.equals(SFTP_PROTOCOL)) {
+        port = DEFAULT_SFTP_PORT;
+      }
+      if (protocol.equals(FTP_PROTOCOL)) {
+        validateFTPConnection(collector, server, port, user, extractedPassword);
+      } else if (protocol.equals(SFTP_PROTOCOL)) {
+        validateSFTPConnection(collector, server, port, user, extractedPassword);
+      }
+    }
+
+    private void validateFTPConnection(FailureCollector collector, String server, int port, String user, String password) {
+      FTPClient ftpClient = new FTPClient();
+      // timeout after 5 seconds
+      ftpClient.setConnectTimeout(5000);
+      try {
+        ftpClient.connect(server, port);
+        try {
+          boolean isLogin = ftpClient.login(user, password);
+          if (!isLogin) {
+            collector.addFailure("Unable to authenticate with given username and password", null)
+                    .withConfigProperty(PATH);
```

Comment:
indent looks incorrect. please format code using this settings file - https://cdap.atlassian.net/wiki/spaces/CE/pages/1594098619/Coding+Standard